### ISSUE TITLE
'first_valid_price' reference before assigntment processor_alpaca.py

### ISSUE THREAD

**Issue by isac997** [AI4Finance-Foundation/FinRL#389]
/FinRL/finrl/finrl_meta/data_processors/processor_alpaca.py", line 86, in clean_data
    tmp_df.iloc[0] = [first_valid_price,
UnboundLocalError: local variable 'first_valid_price' referenced before assignment

seems like it breaks before it has a chance to assign  tmp_df.iloc[0 = [first_valid_price...... ]

please correct me if im am wrong i maybe have misunderstood this
and maybe another solution should be used instead.

but heres my two cents to fix it

change this proccessor_alpaca.py
```
#if the close price of the first row is Nan   
            if str(tmp_df.iloc[0]["close"]) == "nan":
               print('The price of the first row is NaN. It will filled with the first valid price.')
               for i in range(tmp_df.shape[0]):
                   if str(tmp_df.iloc[i]["close"]) != "nan":
                          first_valid_price = tmp_df.iloc[i]["close"]
                          break
                   tmp_df.iloc[0] = [first_valid_price,
                                     first_valid_price,
                                     first_valid_price,
                                     first_valid_price,
                                     0.0,]
```

to this 
```
 #if the close price of the first row is Nan   
            if str(tmp_df.iloc[0]["close"]) == "nan":
               print('The price of the first row is NaN. It will filled with the first valid price.')
               for i in range(tmp_df.shape[0]):
                    if str(tmp_df.iloc[i]["close"]) != "nan":
                        first_valid_price = tmp_df.iloc[i]["close"]
                        tmp_df.iloc[0] = [first_valid_price,
                                     first_valid_price,
                                     first_valid_price,
                                     first_valid_price,
                                     0.0,]
                        break
```

steps to reproduce 
run train.py  with alpaca papertrading
and use START_DATE = 2021-11-23    and END_DATE = 2021-12-07
```
model_type = 'ppo'
    # demo for elegantrl
    train(
        start_date=START_DATE,
        end_date=END_DATE,
        ticker_list=DOW_30_TICKER,
        data_source = 'alpaca',
        time_interval= '1Min', 
        technical_indicator_list= TECHNICAL_INDICATORS_LIST,
        drl_lib='elegantrl', 
        env=env,
        model_name=model_type, 
        API_KEY = API_KEY, 
        API_SECRET = API_SECRET, 
        APCA_API_BASE_URL = APCA_API_BASE_URL,
        erl_params=ERL_PARAMS,
        cwd=f'./papertrading_{model_type}_retrain_latest', #current_working_dir
        break_step=4e5
    )
```

**Comment by benwaldner** [comment]
Do you experience it on a colab file? I had the same problem using colab but could fix it by setting cwd to a fixed path on my Google Drive. Not sure why I came to this conclusion but I think I didnt had this error before.

**Comment by rayrui312** [comment]
Thanks. I think you are right about it. We have updated related codes. Besides what you mentioned, we also add codes to fill the first row with 0 if all the prices are NaN. You may have a check.

**Comment by isac997** [comment]
closing this issue since it was fixed in commit "update alpaca processor" : ddbb9d0fd0c77e9fe4612efd75fc7645a26ee1b7
